yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -412,7 +414,7 @@ private[spark] class Client(
     }
 
     // Distribute an archive with Hadoop and Spark configuration for the AM.
-    val (_, confLocalizedPath) = distribute(createConfArchive().getAbsolutePath(),
+    val (_, confLocalizedPath) = distribute(createConfArchive().toURI.getPath,
```

Comment:
This is a crititcal one, but it doesn't relate to the ticket at all. I get an Exception claiming that it cannot convert a windows path to an URI. In line 320 (where I put those obnoxious debug logs) it fails when using "getAbsolutePath()" here. The proper way would to get a URI and pass that, so that we'll never ever get any complications.